From 3bd46cb9a51773f103ef52b39d6407740eb0d60a Mon Sep 17 00:00:00 2001
From: Eugene Zemtsov <eugene@chromium.org>
Date: Thu, 24 Feb 2022 23:17:20 +0000
Subject: [PATCH] webcodecs: Stop using AudioOpusEncoder as backed for mojo
 audio encoder

AudioOpusEncoder was only used here for testing. Let's not let it get
comfortable. We'll use MF AAC encoder here when we have it. (Soon...)

Bug: 1259883
Change-Id: Ia1819395c8c8fd6d403d4b8558c12f9a1bf7e761
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3489449
Commit-Queue: Eugene Zemtsov <eugene@chromium.org>
Auto-Submit: Eugene Zemtsov <eugene@chromium.org>
Reviewed-by: Dale Curtis <dalecurtis@chromium.org>
Commit-Queue: Dale Curtis <dalecurtis@chromium.org>
Cr-Commit-Position: refs/heads/main@{#974895}
---
 media/mojo/services/gpu_mojo_media_client.cc | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/media/mojo/services/gpu_mojo_media_client.cc b/media/mojo/services/gpu_mojo_media_client.cc
index 8f83a4d6cf6..40cdaff8d3a 100644
--- a/media/mojo/services/gpu_mojo_media_client.cc
+++ b/media/mojo/services/gpu_mojo_media_client.cc
@@ -13,7 +13,6 @@
 #include "build/chromeos_buildflags.h"
 #include "gpu/ipc/service/gpu_channel.h"
 #include "media/audio/audio_features.h"
-#include "media/audio/audio_opus_encoder.h"
 #include "media/base/audio_decoder.h"
 #include "media/base/cdm_factory.h"
 #include "media/base/media_switches.h"
@@ -119,14 +118,7 @@ std::unique_ptr<AudioEncoder> GpuMojoMediaClient::CreateAudioEncoder(
     scoped_refptr<base::SequencedTaskRunner> task_runner) {
   if (!base::FeatureList::IsEnabled(features::kPlatformAudioEncoder))
     return nullptr;
-  // TODO(crbug.com/1259883) Right now Opus encoder is all we have, later on
-  // we'll create a real platform encoder here.
-  auto opus_encoder = std::make_unique<AudioOpusEncoder>();
-  auto encoding_runner = base::ThreadPool::CreateSequencedTaskRunner(
-      {base::TaskPriority::USER_BLOCKING});
-  return std::make_unique<OffloadingAudioEncoder>(std::move(opus_encoder),
-                                                  std::move(encoding_runner),
-                                                  std::move(task_runner));
+  return nullptr;
 }
 
 VideoDecoderType GpuMojoMediaClient::GetDecoderImplementationType() {
